services/web-app/lib/github.js

```diff
@@ -505,68 +546,92 @@ export const getLibraryRelatedLibs = async (libData) => {
-          relatedLibData.assets?.length &&
-          !relatedLibData.assets[0].content?.noIndex &&
-          relatedLibData.assets[0].content?.framework
-        ) {
-          otherAssetFrameworks.push({
-            framework: relatedLibData.assets[0]?.content.framework,
-            params: {
-              library: slug,
-              ...libraryParams,
-              ref: params.ref,
-              asset: params.asset
-            }
-          })
+export const getAssetRelatedFrameworks = withTrace(
+  logging,
+  async function getAssetRelatedFrameworks(params, library) {
+    const otherAssetFrameworks = []
+    if (library.params.group) {
+      for (const [slug, libraryParams] of Object.entries(libraryAllowList)) {
+        if (libraryParams.group === library.params.group) {
+          const libParams = {
+            library: slug,
+            ref: 'latest',
+            ...libraryParams,
+            asset: params.asset
+          }
+          const relatedLibData = await getLibraryData(libParams)
+          if (
+            relatedLibData?.content.id !== library.content.id &&
+            !relatedLibData?.content?.noIndex &&
+            relatedLibData.assets?.length &&
+            !relatedLibData.assets[0].content?.noIndex &&
+            relatedLibData.assets[0].content?.framework
+          ) {
+            otherAssetFrameworks.push({
+              framework: relatedLibData.assets[0]?.content.framework,
+              params: {
+                library: slug,
+                ...libraryParams,
+                ref: params.ref,
+                asset: params.asset
+              }
+            })
+          }
         }
       }
     }
+    return otherAssetFrameworks
   }
-  return otherAssetFrameworks
-}
+)
 
 /**
  * If the params map to a valid design kit in the allowlist, fetch the contents of the design kit's
  * metadata file. If the params are not valid, early return.
  * @param {import('@/typedefs').Params} params
  * @returns {import('@/typedefs').DesignKit[]}
  */
-export const getDesignKitsData = async (params = {}) => {
+export const getDesignKitsData = withTrace(logging, async function getDesignKitsData(params = {}) {
   const designKitsParams = await validateDesignKitsParams(params)
 
-  if (isEmpty(designKitsParams)) return null
+  if (isEmpty(designKitsParams)) {
+    logging.warn(
+      `Skipping design kit with params ${JSON.stringify(
+        params
+      )} because params could not be validated`
+    )
+    // TODO: throw error instead
+    return null
+  }
 
   /**
    * @type {import('@/typedefs').GitHubContentResponse}
    */
   let response = {}
 
+  const options = {
+    owner: designKitsParams.org,
+    repo: designKitsParams.repo,
+    path: removeLeadingSlash(`${designKitsParams.path}/carbon.yml`),
+    ref: designKitsParams.ref
+  }
   try {
+    logging.info(
+      `Attempting to obtain repository's carbon.yml file contents with options: ${JSON.stringify(
+        options
+      )}`
+    )
```

Comment:
As a general comment across this PR, (and I'll add this to my running list of tips for logging), info, warn, and error logs will typically be past-tense, as in "something happened" and you're logging the fact that it did. Debug statements are a bit different in that they can be present or past tense.

For most of the "attempting to..." logs, I'd either switch those to debug statements or move them slightly further down in their respective functions so that they become more of a "I did this thing and here's the result" sort of log.

This might help to consolidate some of the logs you currently have close to each other.